status/condition_set.go

```diff
@@ -209,9 +211,12 @@ func (c ConditionSet) recomputeRootCondition(conditionType string) {
 	} else {
 		c.Set(Condition{
 			Type: c.root,
-			// The root condition is no longer unknown as soon as any are false
+			// The root condition is no longer unknown as soon as any dependency condition false with latest observedGeneration
 			Status: lo.Ternary(
-				lo.ContainsBy(conditions, func(condition Condition) bool { return condition.IsFalse() }),
+				lo.ContainsBy(conditions, func(condition Condition) bool {
+					return condition.IsFalse() &&
+						condition.ObservedGeneration == c.object.GetGeneration()
+				}),
 				metav1.ConditionFalse,
 				metav1.ConditionUnknown,
 			),
```

Comment:
I think we may need some special handling to change the `condition.Status` to unknown for a sub-condition if the current status isn't from the latest generation. Otherwise, you're going to get some weird error messages that say that you have "UnhealthyDependents" but might actually have "True" status conditions for an old generation

Reply:
I wonder if we can create a new Reason value `ReconcilingDependents` for cases when we are Unknown and we are still reconciling the state of other conditions

Reply:
```
I think we may need some special handling to change the condition.Status to unknown for a sub-condition if the current status isn't from the latest generation. Otherwise, you're going to get some weird error messages that say that you have "UnhealthyDependents" but might actually have "True" status conditions for an old generation
```
 this situation is already set to Unknown from above logic?